libs/window-ref/src/lib/window-ref.service.ts

```diff
@@ -19,10 +21,39 @@ export class WindowRefService {
     return this.window.document;
   }
 
+  public getHash() {
+    return this.window.location.hash;
+  }
+
   public getIndexedDB() {
     return this.window.indexedDB;
   }
 
+  public getPlopdownFromHash(): Observable<string> {
+    const plopdowns = new Observable<string>(subscriber => {
```

Comment:
To make sure zone.js wraps the event, you should use:

```ts
@HostListener('window:hashchange', ['$event'])
private function(event) {
    this.hashChange$.next(this.window.location.hash);
}
```

then you can hopefully avoid the change detector later down

Reply:
I wanna test this out myself and see if it does actually get wrapped by zone or not...